src/bgw/scheduler.c

```diff
@@ -245,6 +276,7 @@ scheduled_bgw_job_transition_state_to(ScheduledBgwJob *sjob, JobState new_state)
 					 sjob->job.fd.id);
 				ts_bgw_job_cache_invalidate_callback();
 				CommitTransactionCommand();
+				MemoryContextSwitchTo(scratch_mctx);
```

Comment:
Would be nice to have Assertions for the memory contexts we're on after `StartTransactionCommand` and `CommitTransactionCommand` just to make that clear.

Reply:
i've added it in one place in the main loop